app/services/menu/item.rb

```diff
@@ -27,7 +27,8 @@ def url
     end
 
     def url_hash
-      @url_hash ||= @context.routes.url_helpers.send("hash_for_#{name}_path")
+      return @url_hash if @url_hash
+      @url_hash = @context.routes.url_helpers.send("hash_for_#{name}_path")
       @url_hash.inject({}) do |h,(key,value)|
         h[key] = (value.respond_to?(:call) ? value.call : value)
```

Comment:
Returning early prevents this entire block of code from ever running - so procs aren't executed here.